xbmc/FileItem.cpp

```diff
@@ -133,7 +133,34 @@ CFileItem::CFileItem(const CEpgInfoTagPtr& tag)
   m_bIsFolder = false;
   m_epgInfoTag = tag;
   m_strPath = tag->Path();
-  SetLabel(tag->Title());
+
+  std::string strLabel("");
+  std::string strTitleEpisodeName("");
+  //Title and Episode
+  if (tag->EpisodeName().empty())
+    strTitleEpisodeName = tag->Title();
+  else
+    strTitleEpisodeName = StringUtils::Format("%s (%s)", tag->Title().c_str(), tag->EpisodeName().c_str());
+  //Season + Episode
+  int iSeason(-1);
+  int iEpisode(-1);
+  if (tag->SeriesNumber() > 0)
+    iSeason = tag->SeriesNumber();
+  if (tag->EpisodeNumber() > 0)
+    iEpisode = tag->EpisodeNumber();
+  if (iEpisode >= 0)
+  {
+    if (iSeason == 0) // prefix episode with 'S'
+      strLabel = StringUtils::Format("%s S%d", strTitleEpisodeName.c_str(), iEpisode);
+    else
+      strLabel = StringUtils::Format("%s %dx%d", strTitleEpisodeName.c_str(), iSeason, iEpisode);
```

Comment:
I've done some more digging and this format in confluence at least is actually set by the skin (ViewsVideoLibrary.xml) <code>\<label fallback="416"\>$INFO[listitem.Season,,x]$INFO[listitem.Episode]\</label\></code>